docs/book/src/user/troubleshooting.md

```diff
@@ -38,6 +48,25 @@ For convenience, here is an example one-liner to do this post installation
 kubectl get nodes --no-headers -l '!node-role.kubernetes.io/master' -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}' | xargs -I{} kubectl label node {} node-role.kubernetes.io/worker=''
 ```                  
 
+## Troubleshooting Windows Node Bootstrapping Errors
+
+- Check for errors in the bootstrapper you are using, such as `C:\Program Files\Cloudbase Solutions\Cloudbase-Init\log`.
+- Be sure to look at `DEBUG` logs, as you might have a fatal error hidden in these log messages.  Possible bootstrap errors are:
+  - The windows node is missing some Binaries which you expected to have in the image on boot.
+  - The windows node might be accidentally running powershell or choco commands during bootstrap that aren't installed. 
+
+
+## Getting the bootstrap data manually to inspect it
+
+Ultimately, if a VM isnt joining/bootstrapping into the cluster, you'll probably just want to see wether or not it has the bootstrap data.  This is the most direct way to determine where in the chain of command your bootstrap is failing.
+
+- If your VM doesnt have the right bootstrap data, its because of a bug in the provider's ability to attach bootstrap data.
+- If your VM does have the right bootstrap data, then theres a bug in the way you are configuring its startup logic, and this is most often the case.
+
+Irrespective of OS, on any Vsphere VM, for you can confirm the bootstrap information in the `Configuration Parameters` section of your VSphere machine settings, which are available from the `VM Hardware -> Edit Settings -> Advanced -> Configuration Parameters` section, and then read them by copying them and running `base64 --decode` on their contents.   
+
+Other clouds will have similar methodologies for inspecting the user data attached to a VM.  
```

Comment:
Maybe make the vSphere issue a subsection and open an issue to add the information to troubleshoot this issue for other cloud providers ?